### PR TITLE
Dropping Java support from oneDAL library

### PR DESCRIPTION
**Pull request by napetrov**
# Description
This PR drops Java interfaces support from oneDAL library

- java code
- java build
- Documentation update
- Examples/samples

Java functionality will be migrated to OAP MLlib https://github.com/oap-project/oap-mllib/pull/336

### CODE REVIEW

**Comment by napetrov** [comment]
/intelci: run
set infra_branch=dev/napetrov_java_removal

**Comment by napetrov** [comment]
/intelci: run
set infra_branch=dev/napetrov_java_removal

**Comment by napetrov** [comment]
/intelci: run
set infra_branch=dev/napetrov_java_removal

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ There are different ways for you to build high-performance data science applicat
 - Use oneDAL C++ interfaces with or without SYCL support ([learn more](https://oneapi-src.github.io/oneDAL/#oneapi-vs-daal-interfaces))
 - Use [Intel(R) Extension for Scikit-learn*](https://intel.github.io/scikit-learn-intelex/) to accelerate existing scikit-learn code without changing it
 - Use [daal4py](https://github.com/intel/scikit-learn-intelex/tree/master/daal4py), a standalone package with Python API for oneDAL
-Deprecation Notice: The Java interfaces are deprecated in the oneDAL library and may no longer be supported in future releases.
+Deprecation Notice: The Java interfaces are removed from the oneDAL library.
```
Comment (ethanglaser) [inline README.md:36]:
```suggestion
- Deprecation Notice: The Java interfaces are removed from the oneDAL library.
```

**Comment by ethanglaser** [comment]
Some other possible changes:
Update this similar as done in README: https://github.com/oneapi-src/oneDAL/blob/master/docs/source/deprecation.rst?plain=1#L26

**Review by ethanglaser — APPROVED** [review]

**Comment by napetrov** [comment]
/intelci: run
set infra_branch=dev/napetrov_java_removal

**Comment by napetrov** [comment]
/intelci: run
set infra_branch=dev/napetrov_java_removal

**Pull request merged**